### PR TITLE
new: add dev docs for predicate interface

### PR DESCRIPTION
**Pull request by jdkanani**

### CODE REVIEW

File: contracts/root/TokenPredicates/ITokenPredicate.sol [hunk 1]
```diff
@@ -2,18 +2,36 @@ pragma solidity ^0.6.6;
 
 import {RLPReader} from "../../lib/RLPReader.sol";
 
+/// @title Token predicate interface for all pos portal prdicates
```
Comment (atvanguard) [inline contracts/root/TokenPredicates/ITokenPredicate.sol:5]: nit: typo _prdicates_

File: contracts/root/TokenPredicates/ITokenPredicate.sol [hunk 2]
```diff
@@ -2,18 +2,36 @@ pragma solidity ^0.6.6;
...
         address depositReceiver,
         address rootToken,
         bytes calldata depositData
     ) external;
 
+    /// @notice Validates exit while withdraw process
+    /// @dev Validates exit log while withdrawing. Reverts if validation fails
+    /// @param withdrawer Address who wants to withdraw tokens
+    /// @param logRLPList Log bytes from sidechain
     function validateExitLog(address withdrawer, bytes calldata logRLPList)
```
Comment (atvanguard) [inline contracts/root/TokenPredicates/ITokenPredicate.sol:26]: I spoke to Arth earlier about removing this. So let #15 be merged first - assuming this change is present in that PR.

Comment (arthcp) [reply contracts/root/TokenPredicates/ITokenPredicate.sol:26]: yep removing `lockTokens` and moving its logic to `exitTokens` as part of https://github.com/maticnetwork/pos-portal/pull/15

Comment (arthcp) [reply contracts/root/TokenPredicates/ITokenPredicate.sol]: changed this

**Pull request merged**